src/dmtcp_coordinator.cpp

```diff
@@ -1796,7 +1799,7 @@ int main ( int argc, char** argv )
       setenv(ENV_VAR_CHECKPOINT_DIR, argv[1], 1);
       shift; shift;
     }else if(argc>1 && (s == "-t" || s == "--tmpdir")){
-      setenv(ENV_VAR_TMPDIR, argv[1], 1);
+      tmpDir = Util::calcTmpDir(argv[1]);
```

Comment:
Do we need the tmpdir_arg trick?  This code is inside:  dmtcp_coordinator.cpp:main().
Because it's invoked at the top level function, it should be enough to set the local variable tmpDir, and then pass that value into other functions.